### ISSUE TITLE
Build error on Mac OS X 10.7.3 (Lion)

### ISSUE THREAD

**Issue by kfl** [HIPERFIT/hopencl#9]
When I try to install hopencl with cabal from hackage I get the error:

```
Configuring hopencl-0.2.0...
c2hs: C header contains errors:

/usr/include/stdlib.h:275: (column 20) [ERROR]  >>> Syntax error !
  The symbol `^' does not fit here.

Building hopencl-0.2.0...
Preprocessing library hopencl-0.2.0...
c2hs: Foreign/OpenCL/Bindings/Internal/Types.chi not found in:
dist/build
.
```

The relevant part of `/usr/include/stdlib.h` is

``` .c
#ifdef __BLOCKS__
int  atexit_b(void (^)(void)) __OSX_AVAILABLE_STARTING(__MAC_10_6, __IPHONE_3_2);
void    *bsearch_b(const void *, const void *, size_t,
        size_t, int (^)(const void *, const void *)) __OSX_AVAILABLE_STARTING(__MAC_10_6, __IPHONE_3_2);
#endif /* __BLOCKS__ */
```

**Comment by dybber** [comment]
That's weird. Manuel and I discovered that problem and solved it by undefining **BLOCKS** in the .cabal file, like this:

```
if os(darwin)
  cc-options: "-U__BLOCKS__"
```

This fix is part of the current version on hackage, so I wonder why it doesn't work for you. Maybe the cc-options is no longer passed on to c2hs? 

Try installing it with:

```
cabal install hopencl --c2hs-option=--cppopts=-U__BLOCKS__
```

Related bug report for c2hs: http://hackage.haskell.org/trac/c2hs/ticket/29

**Comment by kfl** [comment]
Passing options with `--c2hs-option` does not seem to help

Installing with verbose flag

```
cabal install -v hopencl --c2hs-option=--cppopts=-U__BLOCKS__
```

Gives the following errors:

```
Reading available packages...
Resolving dependencies...
In order, the following would be installed:
hopencl-0.2.0 (new package)
Extracting
/Users/ken/.cabal/packages/hackage.haskell.org/hopencl/0.2.0/hopencl-0.2.0.tar.gz
to /var/folders/tz/_5z_2zln6fs4s7lcylldkhj80000gn/T/hopencl-0.2.069746...
creating
/var/folders/tz/_5z_2zln6fs4s7lcylldkhj80000gn/T/hopencl-0.2.069746/hopencl-0.2.0/dist/setup
creating
/var/folders/tz/_5z_2zln6fs4s7lcylldkhj80000gn/T/hopencl-0.2.069746/hopencl-0.2.0/dist
creating
/var/folders/tz/_5z_2zln6fs4s7lcylldkhj80000gn/T/hopencl-0.2.069746/hopencl-0.2.0/dist/setup
/usr/local/bin/ghc --make /var/folders/tz/_5z_2zln6fs4s7lcylldkhj80000gn/T/hopencl-0.2.069746/hopencl-0.2.0/Setup.hs -o /var/folders/tz/_5z_2zln6fs4s7lcylldkhj80000gn/T/hopencl-0.2.069746/hopencl-0.2.0/dist/setup/setup -odir /var/folders/tz/_5z_2zln6fs4s7lcylldkhj80000gn/T/hopencl-0.2.069746/hopencl-0.2.0/dist/setup -hidir /var/folders/tz/_5z_2zln6fs4s7lcylldkhj80000gn/T/hopencl-0.2.069746/hopencl-0.2.0/dist/setup -i -i/var/folders/tz/_5z_2zln6fs4s7lcylldkhj80000gn/T/hopencl-0.2.069746/hopencl-0.2.0 -package Cabal-1.14.0
[1 of 1] Compiling Main             ( /var/folders/tz/_5z_2zln6fs4s7lcylldkhj80000gn/T/hopencl-0.2.069746/hopencl-0.2.0/Setup.hs, /var/folders/tz/_5z_2zln6fs4s7lcylldkhj80000gn/T/hopencl-0.2.069746/hopencl-0.2.0/dist/setup/Main.o )
Linking /var/folders/tz/_5z_2zln6fs4s7lcylldkhj80000gn/T/hopencl-0.2.069746/hopencl-0.2.0/dist/setup/setup ...
/var/folders/tz/_5z_2zln6fs4s7lcylldkhj80000gn/T/hopencl-0.2.069746/hopencl-0.2.0/dist/setup/setup
configure --verbose=2 --ghc --prefix=/Users/ken/.cabal --user
--extra-include-dirs=/opt/local/include --extra-lib-dirs=/opt/local/lib
--extra-lib-dirs=/usr/lib --constraint=base ==4.5.0.0 --constraint=bytestring
==0.9.2.1 --c2hs-option=--cppopts=-U__BLOCKS__
Configuring hopencl-0.2.0...
Dependency base ==4.5.0.0: using base-4.5.0.0
Dependency bytestring ==0.9.2.1: using bytestring-0.9.2.1
Using Cabal-1.14.0 compiled by ghc-7.4
Using compiler: ghc-7.4.1

[SNIP]

creating
/var/folders/tz/_5z_2zln6fs4s7lcylldkhj80000gn/T/hopencl-0.2.069746/hopencl-0.2.0/dist/setup
/var/folders/tz/_5z_2zln6fs4s7lcylldkhj80000gn/T/hopencl-0.2.069746/hopencl-0.2.0/dist/setup/setup
build --verbose=2
c2hs: C header contains errors:

/usr/include/stdlib.h:275: (column 20) [ERROR]  >>> Syntax error !
  The symbol `^' does not fit here.

creating dist/build
creating dist/build/autogen
Building hopencl-0.2.0...
Preprocessing library hopencl-0.2.0...
creating dist/build/Foreign/OpenCL/Bindings
/Users/ken/.cabal/bin/c2hs --cpp=/usr/bin/gcc --cppopts=-E --cppopts=-D__GLASGOW_HASKELL__=704 --cppopts=-Ddarwin_BUILD_OS --cppopts=-Ddarwin_HOST_OS --cppopts=-Dx86_64_BUILD_ARCH --cppopts=-Dx86_64_HOST_ARCH --cppopts=-DCALLCONV=ccall --cppopts=-Iinclude --cppopts=-I/opt/local/include --cppopts=-U__BLOCKS__ --include=dist/build --cppopts=-I/usr/local/Cellar/ghc/7.4.1/lib/ghc-7.4.1/bytestring-0.9.2.1/include --cppopts=-I/usr/local/Cellar/ghc/7.4.1/lib/ghc-7.4.1/base-4.5.0.0/include --cppopts=-I/usr/local/Cellar/ghc/7.4.1/lib/ghc-7.4.1/include --output-dir=dist/build --output=Foreign/OpenCL/Bindings/Types.hs ./Foreign/OpenCL/Bindings/Types.chs --cppopts=-U__BLOCKS__
There must be exactly one binding file (suffix .chs),
and optionally one or more header files (suffix .h).
Try the option `--help' on its own for more information.
World file is already up to date.
```

It seems that setup is called without any options?

**Comment by axman6** [comment]
I'm still running into this too. Any news on the issue? I've been trying desperately to get c2hs to pass the -U__BLOCKS__ to gcc, but it refuses to. I can't tell if this is a bug in c2hs or what, but it's certainly amazingly annoying and completely stopping me from helping out with accelerate-opencl.

**Comment by dybber** [comment]
I've written Manuel a mail, we got it running on his Macbook in December. I hope he has got a suggestion. I don't own a Mac myself, so I have a hard time debugging this on my own.

**Comment by axman6** [comment]
Thanks for that. If we can't sort anything out sooner, I'll hopefully be able to talk to Manuel in July at AusHac about it.

**Comment by dybber** [comment]
Should be fixed with #11.
